### PR TITLE
Make it easier to move config

### PR DESCRIPTION
**Pull request by slootjes**
| Q             | A
| ------------- | ---
| License       | MIT

Small change, only removes some duplicate code to make it easier to have a different config dir.

### CODE REVIEW

File: symfony/framework-bundle/3.3/src/Kernel.php [hunk 1]
```diff
@@ -37,25 +47,22 @@ public function registerBundles()
 
     protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader)
     {
-        $container->addResource(new FileResource($this->getProjectDir().'/config/bundles.php'));
+        $container->addResource(new FileResource($this->getBundlesFile()));
         // Feel free to remove the "container.autowiring.strict_mode" parameter
         // if you are using symfony/dependency-injection 4.0+ as it's the default behavior
         $container->setParameter('container.autowiring.strict_mode', true);
         $container->setParameter('container.dumper.inline_class_loader', true);
-        $confDir = $this->getProjectDir().'/config';
```
Comment (Pierstoval) [inline symfony/framework-bundle/3.3/src/Kernel.php]: Instead I'd suggest to only replace this change by this:

```diff
-        $confDir = $this->getProjectDir().'/config';
+        $confDir = $this->getConfigDir();
```
Having a single var reduces the amount of calls, and therefore improves performances 😉

File: symfony/framework-bundle/3.3/src/Kernel.php [hunk 2]
```diff
@@ -37,25 +47,22 @@ public function registerBundles()
...
-        $loader->load($confDir.'/{services}_'.$this->environment.self::CONFIG_EXTS, 'glob');
+        $loader->load($this->getConfigDir().'/{packages}/*'.self::CONFIG_EXTS, 'glob');
+        $loader->load($this->getConfigDir().'/{packages}/'.$this->environment.'/**/*'.self::CONFIG_EXTS, 'glob');
+        $loader->load($this->getConfigDir().'/{services}'.self::CONFIG_EXTS, 'glob');
+        $loader->load($this->getConfigDir().'/{services}_'.$this->environment.self::CONFIG_EXTS, 'glob');
     }
 
     protected function configureRoutes(RouteCollectionBuilder $routes)
     {
-        $confDir = $this->getProjectDir().'/config';
```
Comment (Pierstoval) [inline symfony/framework-bundle/3.3/src/Kernel.php]: Same here: I'd suggest to keep the var for performances

**Comment by slootjes** [comment]
Thanks for the feedback, if Flex doesn't allow to change these paths then indeed it doesn't make sense to make the methods public. Even though the idea behind the PR can't done I do think it's still a small improvement.

File: symfony/framework-bundle/3.3/src/Kernel.php [hunk 3]
```diff
@@ -25,9 +25,19 @@ public function getLogDir()
...
 
+    private function getConfigDir()
+    {
+        return $this->getProjectDir().'/config';
+    }
+
+    private function getBundlesFile()
+    {
+        return $this->getConfigDir().'/bundles.php';
+    }
```
Comment (fabpot) [inline symfony/framework-bundle/3.3/src/Kernel.php]: Can you move these methods at the end of the class (we tend to put private methods after public and protected ones).

File: symfony/framework-bundle/3.3/src/Kernel.php [hunk 4]
```diff
@@ -52,10 +52,19 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
...
         $routes->import($confDir.'/{routes}/'.$this->environment.'/**/*'.self::CONFIG_EXTS, '/', 'glob');
         $routes->import($confDir.'/{routes}'.self::CONFIG_EXTS, '/', 'glob');
     }
+
+    private function getConfigDir()
+    {
+        return $this->getProjectDir().'/config';
+    }
+
+    private function getBundlesFile()
```
Comment (stof) [inline symfony/framework-bundle/3.3/src/Kernel.php]: IMO, this method should not exist at all. Changing its implementation must never happen, as Flex will keep using `%CONFIG_DIR%/bundles.php`. So I would keep `$this->getConfigDir().'/bundles.php'` inline in both places

**Pull request closed without merging**